### DISCUSSION TITLE
Shadow/Directional Lighting Weird Behaviour

### DISCUSSION THREAD

**Discussion by jaalibandar** [cables-gl/cables_docs#327] [! Welcome - readme first!]
Link to the Test Patch: https://cables.gl/edit/ZbMmpb 

Context: 
I am using the render2texture op to render at 4096x4096(Anything fixed but not equal to the viewport dimensions) while using Directional Lighting as a light source to cast shadows using the shadow op.

Problem: 
Enabling Directional Lighting Op > Shadow Map Settings > Rendering Active results in unexpected behaviour.  
Expected behaviour is a rendered view with the shadows working as expected. However enabling the above mentioned parameter is causing some sort of a issue with the view that seems to fix it at viewport dimensions and then it gets scaled in an unexpected manner.
Same behaviour in dev.

Expected View
![image](https://user-images.githubusercontent.com/86233755/186869919-47cda89c-177b-43a2-806f-e56c298a39c8.png)

Unexpected View
![image](https://user-images.githubusercontent.com/86233755/186870040-ba35deee-7cb5-4fa3-8184-242c1d4a3f0f.png)

**Comment by steam0r** [comment]
hey, thanks for reporting. we will look into that, it is cause by having "show selected op helper" enabled. a workaround until the fix would be to disable that option while workin gon these specific ops...

<img width="299" alt="Bildschirmfoto 2022-08-29 um 09 38 18" src="https://user-images.githubusercontent.com/151271/187149148-4a71da5b-a794-4f10-a1d1-ddc674741fc0.png">

**Reply by jaalibandar** [reply]
No, I turned the op helpers off and tried enabling rendering active - the problem persists, doesn't seem to be affected by the op helpers.
![image](https://user-images.githubusercontent.com/86233755/187350051-c07a47d8-7a21-499e-bb55-2ce87fe58d3e.png)

**Comment by pandrr** [comment]
this should now not happen anymore on dev.cables.gl

**Reply by jaalibandar** [reply]
I checked but this is still happening

**Reply by jaalibandar** [reply]
This bug seems to also have propagated else where but I don't seem to be able to figure out a cause https://dev.cables.gl/p/632f040f3893b462f6abe6b9?s=qKvBGISRNrlZvsAt 

Remove the randomcluster op - marked with a comment to see the difference inbetween the expected output and the current output 


buggy output 
![image](https://user-images.githubusercontent.com/86233755/192100413-ab13d325-e8cc-47d9-92b1-ab0900458fb8.png)

expected output 
![image](https://user-images.githubusercontent.com/86233755/192100474-92211930-7c4f-4644-9b06-aa3315474f2d.png)


This is also caused by changing the render2texture size from viewport to manual similar to the issue described above.
